libs/SalesforceSDK/src/com/salesforce/androidsdk/ui/OAuthWebviewHelper.java

```diff
@@ -379,80 +379,90 @@ protected final TokenEndpointResponse doInBackground(RequestType ... params) {
         @Override
         protected void onPostExecute(OAuth2.TokenEndpointResponse tr) {
             final SalesforceSDKManager mgr = SalesforceSDKManager.getInstance();
+
```

Comment:
Looking at failure case first.
Not creating account or saving preferences in case of any failures (exception during flow or managed app restriction).